app/containers/DatasetPage.js

```diff
@@ -332,9 +478,67 @@ class DatasetPage extends Component {
           </FlexBox>
         </FlexBox>
 
+        <FlexBox display="flex" alignItems="center" p={1}
+                 style={uploadBlockStyle}>
+          <FlexBox p={1} flexGrow={1}>
+            <TextField
+              label="File Name"
+              name="uploadFileName"
+              value={this.state.uploadFileName}
+              InputProps={{
+                readOnly: true,
+              }}
+              fullWidth={true} margin="normal"
+            />
+          </FlexBox>
+          <FlexBox p={1}>
+            <input
+              style={{ display: "none" }}
+              id="contained-button-file1"
+              type="file"
+              accept=".yml"
+              onChange={this.handleUpload}
+            />
+            <label htmlFor="contained-button-file1">
+              <Button variant="contained" color="secondary" component="span">
+                Upload yml File
+              </Button>
+            </label>
+          </FlexBox>
+        </FlexBox>
+
+        <FlexBox display="flex" alignItems="center" p={1}
+                 style={fetchBlockStyle}>
+          <FlexBox p={1} flexGrow={1}>
+            <TextField
+              label="URL for the raw file"
+              name="fetchUrl"
+              value={this.state.fetchUrl}
+              onChange={this.handleChangeForm.bind(this)}
+              fullWidth={true} margin="normal" variant="outlined"
+            />
+          </FlexBox>
+          <FlexBox p={1}>
+            <Button variant="contained" color="secondary" component="span"
+            onClick={this.handleFetch}>
+              fetch from this URL
+            </Button>
+          </FlexBox>
+        </FlexBox>
```

Comment:
Because they're used for different tasks - 

- Textfield for file-uploading: this textfield is read-only, which shows the filename for the uploaded file. 
- Textfield for url-fetching: the content is a url input (from the user) where they want to fetch the yml file. 

Please let me know if you have any suggestions on this ^^

Reply:
Yeah I do not know what I meant by this comment, I thought I saw some duplication there but I may have read it wrong! Thank you for explaining it ❤️